x-pack/platform/plugins/shared/actions/server/routes/connector/get_all/get_all.ts

```diff
@@ -34,7 +34,27 @@ export const getAllConnectorsRoute = (
     router.handleLegacyErrors(
       verifyAccessAndContext(licenseState, async function (context, req, res) {
         const actionsClient = (await context.actions).getActionsClient();
-        const result = await actionsClient.getAll();
+        let result = await actionsClient.getAll();
+
+        if (result.some((connector) => connector.actionTypeId === '.inference')) {
+          try {
+            // Get all inference endpoints to filter out inference connectors without endpoints
+            const inferenceEndpoints = await (
+              await context.core
+            ).elasticsearch.client.asInternalUser.inference.get();
```

Comment:
I think we should use the current user (`asCurrentUser`), right?

Reply:
This raises another question: what will we do if the user does not have access to the inference API?

Reply:
It makes sense. Thanks!

Reply:
This should not be fixed in the route, but it should be fixed in the actions client.

Reply:
@cnasikas can you help Dima find the right place here? To make sure that both browser, API, and server-side contract consumers do not see the inference connector if the endpoint is not available.

Reply:
@dgieselaar yes, already discussed with @cnasikas offline, as I found out it is used server-side by other plugins, e.g. https://github.com/elastic/kibana/blob/main/x-pack/solutions/chat/packages/wc-genai-utils/src/connectors/get_connector_list.ts#L25

Reply:
Updated in d79885bbf36b924d39235557a9a25d76811a77e6